designs/aws-launch-template-network-interfaces.md

```diff
@@ -0,0 +1,129 @@
+  AssociateCarrierIpAddress: Boolean
+  AssociatePublicIpAddress: Boolean
+  DeleteOnTermination: Boolean
+  Description: String
+  DeviceIndex: Integer
+  InterfaceType: String
+  NetworkCardIndex: Integer
+  ipv4PrefixCount: Integer
+  ipv6PrefixCount: Integer
+```
+the AWSNodeTemplate will look like this:
+```yaml
+apiVersion: karpenter.k8s.aws/v1alpha1
+kind: AWSNodeTemplate
+metadata:
+  name: default
+spec:
+  subnetSelector: { ... }                 # required, discovers tagged subnets to attach to instances
+  securityGroupSelector: { ... }          # required, discovers tagged security groups to attach to instances
+  instanceProfile: "..."                  # optional, overrides the node's identity from global settings
+  amiFamily: "..."                        # optional, resolves a default ami and userdata
+  amiSelector: { ... }                    # optional, discovers tagged amis to override the amiFamily's default
+  userData: "..."                         # optional, overrides autogenerated userdata with a merge semantic
+  tags: { ... }                           # optional, propagates tags to underlying EC2 resources
+  metadataOptions: { ... }                # optional, configures IMDS for the instance
+  blockDeviceMappings: [ ... ]            # optional, configures storage devices for the instance
+  detailedMonitoring: "..."               # optional, configures detailed monitoring for the instance
+  networkInterfaces:                      # optional, configures network interfaces for the instance
+    - AssociateCarrierIpAddress: false    # optional, Indicates whether to associate a Carrier IP address with eth0 for a new network interface.
+      AssociatePublicIpAddress: false     # optional, Indicates whether to assign a public IPv4 address to eth0 for a new network interface.
+      DeleteOnTermination: false          # optional, Indicates whether the network interface is deleted when the instance is terminated.
+      Description: "..."                  # optional, A description for the network interface.
+      DeviceIndex: 0                      # required, The device index for the network interface attachment.
+      InterfaceType: "interface"          # optional, The interface type for the network interface. To create an Elastic Fabric Adapter (EFA), specify efa
+      NetworkCardIndex: 0                 # optional, The index of the network card. Some instance types support multiple network cards. The primary network interface must be assigned to network card index 0. The default is network card index 0.
+      ipv4PrefixCount: 1                  # optional, The number of IPv4 prefixes that AWS automatically assigns to the network interface.
+      ipv6PrefixCount: 1                  # optional, The number of IPv6 prefixes that AWS automatically assigns to the network interface.
+    - { ... }
+```
+
+in addition, we will only support two networkInterfaces per node, this way we guarantee that the node will be able to
```

Comment:
can we remove this since we're going to leave it up to the user to make sure the instance types they specify can support the number of network interface they specify?